feat: plugins local browser

bugfest opened the pull request:
This PR implements a way to enable plugins in browser for self-hosted LogSeq deployments. It loads the plugins from the local repo. This feature is enabled at build time through the env flag `ENABLE_PLUGINS_LOCAL`.

An alternative `Dockerfile.local` is provided to build the local (cloned) LogSeq repository.

Caveats:
- only plugins not requiring native `electron/ipc` will work  
- plugin updates are not covered by this implementation. Custom docker scripts or using LogSeq app to keep them updated can be easily developed.

Fixes: https://github.com/logseq/logseq/issues/4350, https://github.com/logseq/logseq/issues/6576

Relates with the following entries:
- https://discuss.logseq.com/t/self-hosted-logseq-with-docker-and-plugins-storage/6507
- https://discuss.logseq.com/t/self-hosting-with-plugins-possible/9792/2
- https://discuss.logseq.com/t/enable-plugins-for-self-hosted-logseq/6612

Screenshots:
<img width="480" alt="logseq-selfhosted-local-plugins-1" src="https://github.com/logseq/logseq/assets/52962234/ca30af7a-7857-4cb1-aa35-4589bb3060fb">
<img width="480" alt="logseq-selfhosted-local-plugins-2" src="https://github.com/logseq/logseq/assets/52962234/00f9b768-68e0-43b8-bee4-80a8587b1c75">

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/logseq/logseq?pullRequest=11406) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/logseq/logseq?pullRequest=11406) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/logseq/logseq?pullRequest=11406) it.</sub>

shizzgar commented:
Yo, [bugfest](https://github.com/bugfest), nice work! But unfortunately i can't reproduce it locally.... Could you help a bit? I have cloned your fork, switched on branch with your work, builded docker image using your dockerfile following your instructions, strated Logseq but still cant view page with plugins...

bugfest commented:
> Yo, [bugfest](https://github.com/bugfest), nice work! But unfortunately i can't reproduce it locally.... Could you help a bit? I have cloned your fork, switched on branch with your work, builded docker image using your dockerfile following your instructions, strated Logseq but still cant view page with plugins...

@shizzgar, thanks for the feedback! Happy to see this feature is well received by the community. 

I've tested my PR in an clean environment and didn't have issues. I noticed the instructions to create the `dot-logseq` special directory weren't very clear, plugins needs to be copied to `dot-logseq/plugins/`. I've added more details and fixed a typo in the provided example just in case, please check the updated doc at (https://github.com/bugfest/logseq/blob/feat/plugins-local-browser/docs/docker-web-app-guide.md#with-plugins-enabled).

If you still have issues, can you share your graph directory structure?

shizzgar commented:
@bugfest, thank you, man!
Works like a charm!
